src/pykx/pandas_api/pandas_meta.py

```diff
@@ -210,6 +234,27 @@ def abs(self, numeric_only=False):
             tab = _get_numeric_only_subtable(self)
         return q.abs(tab)
 
+    @api_return
+    def round(self, decimals: Union[int, Dict[str, int]] = 0):
+        tab = self
+        if 'Keyed' in str(type(tab)):
+            tab = q('{(keys x) _ 0!x}', tab)
+
+        return q("""{[t;d]
+            generate_ops:{[vdic]
+                tuples:{flip(2;count[x])#key[x],value[x]}[vdic];
+                key[vdic]!({(({"F"$.Q.f[y]x}[;x[1]])';x[0])}')tuples};
+            get_float_cols:{(key[ct]@where 9=value[ct:abs type each first x])};
```

Comment:
Using `first` to determine the type of a column isn't foolproof as you may be dealing with a column that is a mixed list, for example

```q
([]10?1f;10?1f;1f,9?`a`b`c)
```

This would determine column `x2` to be a float column rather than a mixed list of type `0h`

A more general way of doing this would be the following and it allows for extension to other types (reals in this case)

```q
get_cols:{metaTab:0!meta x;metaTab[`c]where metaTab[`t]in y}
```

This can then be called passing in `"fe"` along with the table to retrieve the float and real columns of the table

```q
get_cols[table;"fe"]
```

Reply:
Please, refer to previous comment.